nomos-da/network/subnetworks-assignations/src/lib.rs

```diff
@@ -31,6 +34,9 @@ pub trait MembershipHandler {
     fn last_subnetwork_id(&self) -> Self::NetworkId;
 
     fn get_address(&self, peer_id: &PeerId) -> Option<Multiaddr>;
+
+    #[must_use]
+    fn rebuild_with(&self, members: Vec<PeerId>, addressbook: HashMap<PeerId, Multiaddr>) -> Self;
```

Comment:
Why this instead of a replacement? I think this is not the way to go probably. The idea was to implement membership for an adapter of the membership service itself. Should be necessary to have a rebuild as it could be done internally.